library/common-library/templates/_agent-config.tpl

```diff
@@ -0,0 +1,30 @@
+{{/*
+This helper should return the defaults that all agents should have
+*/}}
+{{- define "common.agentConfig.defaults" -}}
+{{- if include "common.verboseLog" . }}
+verbose: 1
+{{- end }}
+
+{{- if (include "common.nrStaging" . ) }}
+staging: true
+{{- end }}
+
+{{- with include "common.license._licenseKey" . }}
+{{- /* We will mostly override this using a environment variable but we can have it here just in case */}}
+license: {{ . }}
+{{- end }}
+
+{{- with include "common.proxy" . }}
+proxy: {{ . | quote }}
+{{- end }}
+
+{{- with include "common.fedramp.enabled" . }}
+fedramp: {{ . }}
+{{- end }}
+
+{{- with include "common.customAttributes" . }}
+custom_attributes:
+  {{- . | nindent 2 }}
+{{- end }}
+{{- end -}}
```

Comment:
do we have a test covering the "happy path" of the config generation?

Reply:
For the custom attributes we have it [here](https://github.com/newrelic/helm-charts/blob/common-library/add-agent-global/library/CHART-TEMPLATE/tests/common_library_custom_attributes_test.yaml)

But we don't have it for the config file as you stated above.